ush/forecast_postdet.sh

```diff
@@ -425,7 +449,43 @@ WW3_nml() {
 
 WW3_out() {
   echo "SUB ${FUNCNAME[0]}: Copying output data for WW3"
-  # TODO: Need to add logic to copy restarts from DATArestart/WW3_RESTART to COMOUT_WAVE_RESTART
+
+  # Copy wave namelist from DATA to COMOUT_CONF after the forecast is run (and successfull)
+  ${NCP} "${DATA}/ww3_shel.nml" "${COMOUT_CONF}/ufs.ww3_shel.nml"
+
+  # Copy WW3 restarts at the end of the forecast segment to COM for RUN=gfs|gefs
+  if [[ "${COPY_FINAL_RESTARTS}" == "YES" ]]; then
+    local seconds source_file target_file
+    if [[ "${RUN}" == "gfs" || "${RUN}" == "gefs" ]]; then
+      echo "Copying WW3 restarts for 'RUN=${RUN}' at ${forecast_end_cycle}"
+      target_file="${forecast_end_cycle:0:8}.${forecast_end_cycle:8:2}0000.restart.ww3.nc"
+      ${NCP} "${DATArestart}/WW3_RESTART/${target_file}" \
+             "${COMOUT_WAVE_RESTART}/${target_file}"
+    fi
+  fi
+
+  # Copy restarts for next cycle for RUN=gdas|gefs
+  if [[ "${RUN}" == "gdas" || "${RUN}" == "gefs" ]]; then
+    local restart_date
+    restart_date="${model_start_date_next_cycle}"
+    echo "Copying WW3 restarts for 'RUN=${RUN}' at ${restart_date}"
+    #seconds=$(to_seconds "${restart_date:8:2}0000")  # convert HHMMSS to seconds
+    #source_file="ufs.cpld.ww3.r.${restart_date:0:4}-${restart_date:4:2}-${restart_date:6:2}-${seconds}.nc"
+    target_file="${restart_date:0:8}.${restart_date:8:2}0000.restart.ww3.nc"
+    ${NCP} "${DATArestart}/WW3_RESTART/${target_file}" \
+           "${COMOUT_WAVE_RESTART}/${target_file}"
+  fi
+
+  #Copy restarts for downstream usage in HAFS 
+  if [[ "${RUN}" =~ "gdas" ]]; then
```

Comment:
Just verifying, this is required for both the deterministic and ensemble?

Reply:
The downstream HAFS dependency is only for this specific GDAS file.

Reply:
In  that case, I suggest specifying `RUN=="gdas"`, as `=~` will also copy the wave restarts for the `enkfgdas` `RUN`.

```suggestion
  if [[ "${RUN}" == "gdas" ]]; then
```